projects/sound-open-firmware/Dockerfile

```diff
@@ -20,39 +20,50 @@
 
 FROM gcr.io/oss-fuzz-base/base-builder
 
-RUN apt-get -y update && \
-	    apt-get install -y \
-	    	autoconf \
-		bison \
-		build-essential \
-		flex \
-		gawk \
-		gettext \
-		git \
-		gperf \
-		help2man \
-		libncurses5-dev \
-		libssl-dev \
-		libtool \
-		libtool-bin \
-		pkg-config \
-		software-properties-common \
-		texinfo \
-		udev \
-		cmake \
-		libglib2.0-dev
-
-ARG CLONE_DEFAULTS="--depth 5"
-
-# Use ToT alsa utils for the latest topology patches.
-RUN cd /tmp && \
-git clone $CLONE_DEFAULTS https://github.com/thesofproject/alsa-lib.git && \
-git clone $CLONE_DEFAULTS https://github.com/thesofproject/alsa-utils.git && \
-cd /tmp/alsa-lib && ./gitcompile &&  make install && \
-cd /tmp/alsa-utils && ./gitcompile &&  make install
-
-ARG GITHUB_SOF=https://github.com/thesofproject
-
-RUN cd $SRC && git clone $CLONE_DEFAULTS $GITHUB_SOF/sof
-WORKDIR sof
+# Base packages
+RUN apt-get -y update
+RUN apt-get install -y \
+      gettext git libc6-dev-i386 libglib2.0-dev libncurses5-dev \
+      libtool ninja-build python3-pip
+RUN pip3 install west
+
+# Zephyr SDK:
+#
+# Zephyr doesn't provide a clean "get the latest version" URL, but
+# note that the use of the /latest/ path component at least ensures
+# this will fail on a new release.
+ARG SDK_VER=0.16.1
+WORKDIR /root
+RUN curl -L -o sdktmp.tar.xz https://github.com/zephyrproject-rtos/sdk-ng/releases/latest/download/zephyr-sdk-${SDK_VER}_linux-x86_64_minimal.tar.xz
+RUN tar xf sdktmp.tar.xz; rm sdktmp.tar.xz
+RUN zephyr-sdk-*/setup.sh -h
+
+WORKDIR $SRC
+
+# SOF itself (also pulls in Zephyr + modules via west)
+RUN west init -m https://github.com/thesofproject/sof sof
```

Comment:
West is a git wrapper, analogous to repo.  It's possible, but error prone, to manually clone all the relevant repositories.  The structure isn't complicated, $SRC/sof is the top level directory which contains $SRC/sof/sof, which is the git repo corresponding to the project itself, and $SRC/sof/zephyr, and a few other bits under $SRC/sof/modules/*.

What are the requirements here?  Can I symlink you something to the top level that matches what you want to see?

Reply:
OK, I see it.  srcmap assumes that the git archive was pulled from a remote with the name "origin".  This is true by default with an unadorned git clone.  It's not required, and in fact west allows multiple remotes with names like "thesofproject" or "upstream". 

FWIW: a straightforward fix to srcmap, which replaces the assumption that "there is always a remote named origin" with "we only care about the first remote listed" is something like (untested):

```diff
--- a/infra/base-images/base-builder/srcmap
+++ b/infra/base-images/base-builder/srcmap
@@ -35,7 +35,7 @@ fi
 for DOT_GIT_DIR in $(find $PATHS_TO_SCAN -name ".git" -type d); do
   GIT_DIR=$(dirname $DOT_GIT_DIR)
   cd $GIT_DIR
-  GIT_URL=$(git config --get remote.origin.url)
+  GIT_URL=$(git config --get remote.$(git remote | head -1).url)
   GIT_REV=$(git rev-parse HEAD)
   jq_inplace $SRCMAP ".\"$GIT_DIR\" = { type: \"git\", url: \"$GIT_URL\", rev: \"$GIT_REV\" }"
 done
```

It's worth pointing out that these are still just assumptions.  In the general case it's not possible to look at a .git directory and ask the question "from where was this pulled?".  Maybe it would be better to allow projects to specify the srcmap themselves for edge cases like this?

Is there a way for me to run srcmap locally in a way that corresponds to your build failures to test this?  I'm digging around and getting a little lost.

Reply:
I think the origin trick would work. If you're using the `infra/helper.py shell` command to build/debug, you can run srcmap to figure out if this will work or not.

Reply:
Global fix up at #10383 

Will look at the project-side workaround now.  Honestly didn't even know about west list, but that's cleaner than `find $SRC -name .git` for sure.

Reply:
Project-local fix in #10385

Reply:
Continued in:
- https://github.com/google/oss-fuzz/pull/10385